contracts/Registry.vy

```diff
@@ -654,6 +783,11 @@ def _get_new_pool_coins(
             self.markets[key][length] = _pool
             self.market_counts[key] = length + 1
 
+            # register the coin pair
+            if length == 0:
+                self._register_coin_pair(coin_list[i], coin_list[x])
+                self._register_coin_pair(coin_list[x], coin_list[i])
```

Comment:
Should be possible to use `bitwise_xor`  in `self.coin_swap_indexes` to reduce the number of storage writes, and handle all of this in a single call.